From 9f148d41e0cb5839e8680bd3b4c4bc21510e444b Mon Sep 17 00:00:00 2001
From: Phil Steitz <phil.steitz@gmail.com>
Date: Thu, 23 Jul 2015 20:21:38 -0400
Subject: [PATCH] Changed data size check to be positive length, not > 1 to fix
 ResizableDoubleArray constructor failure on input array of length 1.

JIRA: MATH-1252
Thanks to John Bay
---
 src/changes/changes.xml                                   | 3 +++
 .../apache/commons/math3/util/ResizableDoubleArray.java   | 2 +-
 .../commons/math3/util/ResizableDoubleArrayTest.java      | 8 ++++++++
 3 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/src/changes/changes.xml b/src/changes/changes.xml
index 066aa0f953..d613eded9e 100644
--- a/src/changes/changes.xml
+++ b/src/changes/changes.xml
@@ -51,6 +51,9 @@ If the output is not quite correct, check for invisible trailing spaces!
   </properties>
   <body>
     <release version="3.6" date="XXXX-XX-XX" description="">
+      <action dev="psteitz" type="fix" issue="MATH-1252 due-to="John Bay">
+        ResizableDoubleArray constructor does not work with double array of size 1.
+      </action>
       <action dev="erans" type="fix" issue="MATH-1251">
         Fixed initial value of "number of calls" counter in class "KohonenUpdateAction"
         (package "o.a.c.m.ml.neuralnet.sofm").
diff --git a/src/main/java/org/apache/commons/math3/util/ResizableDoubleArray.java b/src/main/java/org/apache/commons/math3/util/ResizableDoubleArray.java
index 468ed74ca5..dae2b3b646 100644
--- a/src/main/java/org/apache/commons/math3/util/ResizableDoubleArray.java
+++ b/src/main/java/org/apache/commons/math3/util/ResizableDoubleArray.java
@@ -411,7 +411,7 @@ public ResizableDoubleArray(int initialCapacity,
         numElements = 0;
         startIndex = 0;
 
-        if (data != null && data.length > 1) {
+        if (data != null && data.length > 0) {
             addElements(data);
         }
     }
diff --git a/src/test/java/org/apache/commons/math3/util/ResizableDoubleArrayTest.java b/src/test/java/org/apache/commons/math3/util/ResizableDoubleArrayTest.java
index 6d9d2444ce..3a1a1d7765 100644
--- a/src/test/java/org/apache/commons/math3/util/ResizableDoubleArrayTest.java
+++ b/src/test/java/org/apache/commons/math3/util/ResizableDoubleArrayTest.java
@@ -125,6 +125,14 @@ public void testConstructors() {
         ResizableDoubleArray copyDa = new ResizableDoubleArray(testDa);
         Assert.assertEquals(copyDa, testDa);
         Assert.assertEquals(testDa, copyDa);
+        
+        // JIRA: MATH-1252
+        final double[] values = {1};
+        testDa = new ResizableDoubleArray(values);
+        Assert.assertArrayEquals(values, testDa.getElements(), 0);
+        Assert.assertEquals(1, testDa.getNumElements());
+        Assert.assertEquals(1, testDa.getElement(0), 0);
+        
     }
 
 
